src/OpenIddict.EntityFramework/Stores/OpenIddictApplicationStore.cs

```diff
@@ -147,20 +168,37 @@ public class OpenIddictApplicationStore<TApplication, TContext, TKey> : IOpenIdd
         }
 
         /// <summary>
-        /// Retrieves the hashed secret associated with an application.
+        /// Retrieves the token identifiers associated with an application.
         /// </summary>
         /// <param name="application">The application.</param>
         /// <param name="cancellationToken">The <see cref="CancellationToken"/> that can be used to abort the operation.</param>
         /// <returns>
         /// A <see cref="Task"/> that can be used to monitor the asynchronous operation,
-        /// whose result returns the hashed secret associated with the application.
+        /// whose result returns the tokens associated with the application.
         /// </returns>
-        public virtual Task<string> GetHashedSecretAsync(TApplication application, CancellationToken cancellationToken) {
+        public virtual async Task<IEnumerable<string>> GetTokensAsync(TApplication application, CancellationToken cancellationToken) {
             if (application == null) {
                 throw new ArgumentNullException(nameof(application));
             }
 
-            return Task.FromResult(application.Secret);
+            // Ensure that the key type can be serialized.
+            var converter = TypeDescriptor.GetConverter(typeof(TKey));
+            if (!converter.CanConvertTo(typeof(string))) {
+                throw new InvalidOperationException($"The '{typeof(TKey).Name}' key type is not supported.");
+            }
+
+            var query = from entity in Applications
```

Comment:
nit: we use Linq method syntax in most places, is there any specific reason to use the other syntax here?